src/macro_util.rs

```diff
@@ -479,25 +509,83 @@ where
 ///
 /// [`is_bit_valid`]: TryFromBytes::is_bit_valid
 #[inline(always)]
-pub fn try_transmute<Src, Dst>(mut src: Src) -> Result<Dst, ValidityError<Src, Dst>>
+pub fn try_transmute<Src, Dst>(src: Src) -> Result<Dst, ValidityError<Src, Dst>>
 where
     Src: IntoBytes,
     Dst: TryFromBytes,
 {
-    if !is_src_valid::<Src, Dst, _, BecauseExclusive>(Ptr::from_mut(&mut src)) {
-        return Err(ValidityError::new(src));
+    let mut src = ManuallyDrop::new(src);
+    let ptr = Ptr::from_mut(&mut src);
+    match try_cast_or_pme::<_, ManuallyDrop<Unalign<Dst>>, _, BecauseExclusive>(ptr) {
```

Comment:
Can you add a comment explaining why this is guaranteed not to fail due to increased alignment?